frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/disk-inventory.spec.ts

```diff
@@ -0,0 +1,121 @@
+  cy.exec(`kubectl get --ignore-not-found localvolumediscoveryresults -n openshift-storage -A -o json`);
+
+describe('Disk list is accessible from Nodes view', () => {
+  const diskStatusCount = {
+    available: 0,
+    notAvailable: 0,
+    unknown: 0,
+  };
+
+  const localvolumediscoveryresults = {
+    name: null,
+    nodeName: null,
+    discoveredDevices: null,
+  };
+
+  const lvdCount = (lvd) => {
+    if (!lvd) return diskStatusCount;
+    for (var i = 0; i < lvd.length; i++) {
+      switch(lvd[i].status.state) {
+        case "Available":
+          diskStatusCount.available += 1;
+          break;
+        case "NotAvailable":
+          diskStatusCount.notAvailable += 1;
+          break;
+        case "Unknown":
+          diskStatusCount.unknown += 1;
+      }
+    }
+    return diskStatusCount;
+  }
+
+  const toggleDiskSelect = (diskStatus: string) => {
+    cy.byLegacyTestID('filter-dropdown-toggle').click();
+    cy.byTestRowFilter(diskStatus).parent().click();
+  };
+
+  const testDiskSelection = (diskStatus: string) => {
+    var count = 0;
+    toggleDiskSelect(diskStatus);
+    switch(diskStatus) {
+      case "Available":
+        count = diskStatusCount.available;
+        break;
+      case "NotAvailable":
+        count = diskStatusCount.notAvailable;
+        break;
+      case "Unknown":
+        count = diskStatusCount.unknown;
+    }
+    if (count == 0) {
+      cy.byTestRows("resource-row").should('not.exist')
+    } else {
+      cy.byTestRows("resource-row").parent().find('tr')
+        .should('have.length', count)
+    }
+    toggleDiskSelect(diskStatus);
+  }
+
+  before(() => {
+    cy.login();
+    cy.visit('/');
+    cy.clickNavLink(['Compute', 'Nodes']);
+    fetchLocalVolumeDiscovery().then((res) => {
+      const json = JSON.parse(res.stdout);
+      localvolumediscoveryresults.name = json.items[0].metadata.name;
+      localvolumediscoveryresults.nodeName = json.items[0].spec.nodeName;
+      localvolumediscoveryresults.discoveredDevices = json.items[0].status.discoveredDevices;
+    });
+  });
+
+  after(() => {
+    cy.logout();
+  });
+
+  it('`Disks` tab is enabled and accessible', () => {
+    cy.byLegacyTestID(localvolumediscoveryresults.nodeName).click();
+  });
+
+  it('List title is `Disks`', () => {
+    cy.byLegacyTestID('horizontal-link-Disks').contains('Disks');
+    cy.byLegacyTestID('horizontal-link-Disks').click();
+  });
+
+  it('Rows are displayed with correct data format', () => {
+    const devId = localvolumediscoveryresults.discoveredDevices[0].deviceID;
+    cy.byTestRows("resource-row").parent()
+      .find('tr')
+      .should('have.length',
+        localvolumediscoveryresults.discoveredDevices.length)
+    cy.byDataID(devId).contains('td',
+      localvolumediscoveryresults.discoveredDevices[0].path).should('be.visible');
+    cy.byDataID(devId).contains('td',
+      localvolumediscoveryresults.discoveredDevices[0].status.state).should('be.visible')
```

Comment:
Why `should('be.visible')` ? Is the test fails for some input ?

Reply:
Simply I have added this to make sure the path and status are visible. It is not required i feel. I will remove this "be.visible" ...